include/arch/arm/arch/32/mode/object/structures.h

```diff
@@ -99,18 +99,21 @@ typedef struct asid_pool asid_pool_t;
 
 #define HW_ASID_SIZE_BITS 1
 
-#define ASID_POOL_INDEX_BITS seL4_ASIDPoolIndexBits
 #define ASID_BITS (asidHighBits+asidLowBits)
 
 #define nASIDPools BIT(asidHighBits)
 
 #define ASID_LOW(a) (a & MASK(asidLowBits))
-#define ASID_HIGH(a) ((a >> asidLowBits) & MASK(asidHighBits))
+/* This *could* have been defined as ((a >> asidLowBits) & MASK(asidHighBits))
+ * but the compiler does not have enough information to be able to elide the
+ * mask operation, and ASIDs are only ever used internally to the kernel.
+ */
```

Comment:
Comment is misleading as it implies that the compiler would be allowed to elide the mask.

Before, top bits not part of the ASID were ignored, now we assume we only get a complete and valid (unsigned) ASID value passed, so the behaviour has actually changed.